scripts/convert-svgs-to-icon-mixins.js

```diff
@@ -0,0 +1,59 @@
+const fs = require('fs');
+const path = require('path');
+const mime = require('mime');
+const SVGO = require('svgo');
+
+const args = process.argv;
+const symbols = /[\r\n%#()<>?\[\\\]^`{|}]/g;
+const svgo = new SVGO();
+
+function encodeSVG(data) {
+  // Use single quotes instead of double to avoid encoding.
+  data = data.replace(/"/g, "'");
+
+  data = data.replace(/>\s{1,}</g, '><');
+  data = data.replace(/\s{2,}/g, ' ');
+  data = data.replace(symbols, encodeURIComponent);
+  data = data.replace(/fill='%23(.*)' /g, "fill='#{vf-url-friendly-color($color)}' ");
+
+  return data;
+}
+
+function convertSVGs(directory) {
+  fs.readdir(directory, (err, files) => {
+    if (err) {
+      throw err;
+    }
+
+    files.forEach((file) => {
+      const filePath = `${directory}${file}`;
+      const fileType = mime.getType(filePath);
+      const isSVG = fileType === 'image/svg+xml';
+
+      if (isSVG) {
+        fs.readFile(filePath, 'utf-8', function (err, data) {
+          if (err) {
+            throw err;
+          }
+
+          svgo.optimize(data, {path: filePath}).then(function (result) {
+            console.log(`@mixin vf-icon-${path.parse(file).name}($color) {`);
+            console.log(`background-image: url("data:image/svg+xml,${encodeSVG(result.data)}");`);
+            console.log('}');
+          });
+        });
+      }
+    });
+  });
+}
+
+if (args.includes('--directory')) {
```

Comment:
If we are expecting only one argument we could read it without expecting `--directory`.

`dotrun svgs-to-mixins ./my-icons`

Also, just for being a bit more error proof we could verify is the given path exists and is a directory.